### PR TITLE
added ability to do /quote #channel

### PR DESCRIPTION
**Pull request by Thirdegree**
e.g. /quote #nsfw

### CODE REVIEW

**Review by mutoid — CHANGES_REQUESTED** [review]
Rebase and also add error for when no results are found. See final else.

**Comment by Thirdegree** [comment]
Changes made

File: pin_logic.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ def quote
         Chat.new(@channel_id).chat_out(pin.format)
     elsif @query[0] == '#'
         chan_id = Pin.pluck(:channel_name, :channel_id).uniq.to_h[name.gsub(/#/,'')]
+        return "No such channel exists" if !chan_id
         pin = Pin.joins(:author).where(channel_id: chan_id).sample
         Chat.new(chan_id).chat_out(pin.format)
```
Comment (mutoid) [inline pin_logic.rb]: also no pin found error here

**Review by mutoid — CHANGES_REQUESTED** [review]

**Comment by Thirdegree** [comment]
added

**Review by mutoid — APPROVED** [review]

**Pull request merged**